code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,254 @@
+		for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+			animate(L)
+			L.transform = matrix()
+			L.icon_state = initial(L.icon_state)
+			L.update_o()
+		C.do_smoothing = 1
+	else
+		if(new_parallax_movedir == 4 || new_parallax_movedir == 1)
+			C.looping_mode = 1
+		else
+			C.looping_mode = 2
+		for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+			if(new_parallax_movedir == 1 || new_parallax_movedir == 2)
+				L.icon_state = "[initial(L.icon_state)]_vertical"
+			else
+				L.icon_state = "[initial(L.icon_state)]_horizontal"
+			L.update_o()
+			var/T = 50 / L.speed
+			var/matrix/newtransform
+			switch(new_parallax_movedir)
+				if(1)
+					newtransform = matrix(1, 0, 0, 0, 1, 480)
+				if(2)
+					newtransform = matrix(1, 0, 0, 0, 1,-480)
+				if(4)
+					newtransform = matrix(1, 0, 480, 0, 1, 0)
+				if(8)
+					newtransform = matrix(1, 0,-480, 0, 1, 0)
+			L.transform = newtransform
+			animate(L, transform = matrix(), time = T, loop = -1, flags = ANIMATION_END_NOW)
+		C.do_smoothing = 0
+
+	C.parallax_movedir = new_parallax_movedir
+
+/datum/hud/proc/update_parallax()
+	var/client/C = mymob.client
+
+	var/turf/posobj = get_turf(C.eye)
+	var/area/areaobj = posobj.loc
+	set_parallax_movedir(areaobj.parallax_movedir) // Update the movement direction of the parallax if necessary
+
+	if(!C.previous_turf || (C.previous_turf.z != posobj.z))
+		C.previous_turf = posobj
+
+	//Doing it this way prevents parallax layers from "jumping" when you change Z-Levels.
+	var/offset_x = posobj.x - C.previous_turf.x
+	var/offset_y = posobj.y - C.previous_turf.y
+
+	C.previous_turf = posobj
+
+	var/last_delay = 2
+	if(offset_x != 0 || offset_y != 0)
+		var/world_time = world.time
+		last_delay = world_time - C.last_parallax_shift
+		last_delay = min(last_delay, 2)
+		C.last_parallax_shift = world_time
+
+	for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+		L.offset_x -= offset_x * L.speed
+		L.offset_y -= offset_y * L.speed
+
+		if(C.looping_mode == 0)
+			if(L.offset_x > 240)
+				L.offset_x -= 480
+			if(L.offset_x < -240)
+				L.offset_x += 480
+			if(L.offset_y > 240)
+				L.offset_y -= 480
+			if(L.offset_y < -240)
+				L.offset_y += 480
+		else if(C.looping_mode == 1)
+			if(L.offset_x > 0)
+				L.offset_x -= 480
+			if(L.offset_x < -480)
+				L.offset_x += 480
+			if(L.offset_y > 0)
+				L.offset_y -= 480
+			if(L.offset_y < -480)
+				L.offset_y += 480
+		else if(C.looping_mode == 2)
+			if(L.offset_x >= 480)
+				L.offset_x -= 480
+			if(L.offset_x < 0)
+				L.offset_x += 480
+			if(L.offset_y >= 480)
+				L.offset_y -= 480
+			if(L.offset_y < 0)
+				L.offset_y += 480
+
+		if(C.do_smoothing && (offset_x != 0 || offset_y != 0) && (offset_x == 1 || offset_x == -1 || offset_y == 1 || offset_y == -1))
+			L.transform = matrix(1, 0, offset_x*L.speed, 0, 1, offset_y*L.speed)
+			animate(L, transform=matrix(), time = last_delay, flags = ANIMATION_END_NOW)
+
+		L.screen_loc = "CENTER-7:[L.offset_x],CENTER-7:[L.offset_y]"
```

Comment:
What a weird fucking way to do this.

Logic: Need to move something from matrix(1, 0, offsetx, 0, 1, y) to matrix(1, 0, offsetx+1, 0, 1, y). Instead of just changing the matrix into position, the code changes the screen loc directly to where it should be and gives it a NEGATIVE offset to force it back into place and then animates the removal of said negative offset to offset the second offset it just offset the parallax by.